### PR TITLE
fix: `Input` icons logic

### PR DESCRIPTION
**Pull request by snqb**
image

### CODE REVIEW

File: src/components/icon/GetIcon.tsx [hunk 1]
```diff
@@ -19,19 +19,11 @@ const GetIcon: FC<GetIconProps> = ({
 }) => {
   const theme = useTheme();
   const themeColor = path<string>(split('.', color), theme.colors)!;
+
   return (
-    <Flex
-      sx={{
-        svg: {
-          width: prop(size, theme.iconSizes),
-          height: prop(size, theme.iconSizes),
```
Comment (snqb) [inline src/components/icon/GetIcon.tsx]: its set inside `getIcon` function

File: src/components/input/index.tsx [hunk 2]
```diff
@@ -98,7 +98,7 @@ const Input: FC<InputProps> = forwardRef(
             placeholder={placeholder}
             {...props}
           />{' '}
-          {icon && <GetIcon icon={icon} color="gray" size="md" sx={getIconStyle}/>}
+          {icon && <GetIcon icon={icon} color="gray" size="sm" sx={getIconStyle}/>}
```
Comment (snqb) [inline src/components/input/index.tsx:101]: that was too big, changed per request

File: src/components/input/input.stories.tsx [hunk 3]
```diff
@@ -107,53 +115,35 @@ const argTypes = {
     },
   },
   icon: {
+    options: Object.values(IconName).filter((x) => typeof x === 'string'),
+    mapping: IconName,
```
Comment (snqb) [inline src/components/input/input.stories.tsx:119]: now users can select any icon in a dropdown

File: src/components/icon/GetIcon.tsx (hunk 1 above)
Comment (esonhey) [reply src/components/icon/GetIcon.tsx]: it's better to have the correct height/width ratio in the SVG components and here we just set the width as size, and the height to be auto-scale.
if the viewBox prop of SVG is correct then with auto-scaling you can achieve the behavior to have a ratio of 1.

**Review by esonhey — CHANGES_REQUESTED** [review]
Not all the SVGs scale properly on size change.
I think it's better to have the size property just on the width and make the height auto
```jsx
<svg
  width={size}
  height="auto"
  viewBox="0 0 24 24" // 24/24 ratio of width/size
```
Screen Shot 2022-04-21 at 8 47 56 AM

**Comment by snqb** [comment]
@ehsan-github this was causing some problems in `position: absolute` layouts rendering with 0 height.

The assumption is that we have 1:1 icons. Can't see a problem here. Maybe I'm missing something

**Comment by esonhey** [comment]
> @ehsan-github this was causing some problems in `position: absolute` layouts rendering with 0 height.
> 
> The assumption is that we have 1:1 icons. Can't see a problem here. Maybe I'm missing something

Hmmm Ok, then just the issue with the svgs that has fixed width and height not using `size` prop

**Review by esonhey — APPROVED** [review]
Looks good!

**Pull request merged**